src/components/layouts/php-collection-page-layout.tsx

```diff
@@ -0,0 +1,689 @@
+  return (
+    <div key={module.id}>
+      <h2 className="text-xl font-bold mt-4 mb-2 sm:mx-0 mx-auto w-fit">
+        {module.title}
+      </h2>
+      <div>
+        <p className="mb-6 prose text-gray-900 dark:prose-dark md:prose-lg md:dark:prose-lg-dark dark:text-gray-100 dark:prose-a:text-blue-300 dark:hover:prose-a:text-blue-200 prose-a:text-blue-500 hover:prose-a-:text-blue-600 ">
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
+          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+          aliquip ex ea commodo consequat. Duis aute irure dolor in
+          reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
+          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
+          culpa qui officia deserunt mollit anim id est laborum.
+        </p>
+        <ul
+          className={`grid sm:grid-rows-${Math.ceil(
+            module.resourceList.length / 2,
+          )} sm:grid-flow-col gap-4`}
+        >
+          {children}
+        </ul>
+      </div>
+    </div>
+  )
+}
+
+const ResourceCollection = (props: any) => {
+  return props.resourceList.map((resource: any, index: number) => {
+    switch (resource.type) {
+      case 'module':
+        return (
+          <ModuleCollection module={resource}>
+            <ResourceCollection
+              {...props}
+              resourceList={resource.resourceList}
+            />
+          </ModuleCollection>
+        )
+      case 'lesson':
+        return (
+          <LessonLinkResource
+            lesson={resource}
+            index={index}
+            completedLessonSlugs={props.completedLessonSlugs}
+          />
+        )
+      default:
+        return null
+    }
+  })
+}
+
+const PhpCollectionPageLayout: React.FunctionComponent<CoursePageLayoutProps> =
+  ({lessons = [], course, ogImageUrl}) => {
+    const courseDependencies: any = getDependencies(course.slug)
+    const [isFavorite, setIsFavorite] = React.useState(false)
+    const [clickable, setIsClickable] = React.useState(true)
+
+    // Manually slicing lessons into modules
+    const resourceCollection = [
+      {
+        type: 'module',
+        title: 'Project Setup',
+        resourceList: lessons.slice(0, 3),
+      },
+      {type: 'module', title: 'Tags', resourceList: lessons.slice(3, 7)},
+      {
+        type: 'module',
+        title: 'Variables and Constants',
+        resourceList: lessons.slice(7, 15),
+      },
+      {
+        type: 'module',
+        title: 'Conditionals',
+        resourceList: lessons.slice(15, 21),
+      },
+      {
+        type: 'module',
+        title: 'Arrays and Loops',
+        resourceList: lessons.slice(21, 29),
+      },
+      {type: 'module', title: 'Functions', resourceList: lessons.slice(29, 37)},
+      {type: 'module', title: 'Classes', resourceList: lessons.slice(37, 46)},
+    ]
```

Comment:
nice! Is the idea that if we like this pattern we could eventually encode this shape of data in Sanity?

Reply:
Yeah this is the idea! All collections should have a "resourceList" array field